### ISSUE TITLE
glob should results on only directories when used with `*/`

### ISSUE THREAD

**Issue by phcerdan** [fsspec/s3fs#712]

**Comment by martindurant** [comment]
Agree, that glob("...*/") should only return keys ending with "/". These are not necessarily "folders" (any key name is valid in s3). The question of whether a path that corresponds to a common prefix (this is aws's nomenclature) should match with or without the "/" is tricky to answer and implement, since it can be used in both variants is different places; e.g., you would expect ls("bucket/path") and ls("bucket/path/") to give the same results.


> s3fs.glob("/path/*") should show keys for folders ending with "/", and keys for files without ending "/", for consistency with [aws docs.](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/using-folders.html)

Disagree with this: s3fs should show the actual names of the keys it finds. The link here is to the aws console, which has it's own ideas about folders over and above what the actual s3 service and API supports. If you want to know if an entry is a regular file, use `detail=True`.
